modules/document_repository/php/files.class.inc

```diff
@@ -88,7 +88,6 @@ class Files extends \NDB_Page
                     )
                 );
             }
-            break;
```

Comment:
These ones are scary.. do we have any other static analysis rules to warn about unexpected fallthroughs if the if statements change?

Reply:
Not that I know of... I think it'd be tricky because PHP doesn't make you do `break` at the end and we take advantage of empty fallthrough cases quite a bit, like in LINST parsing.

I could add a suppression for the unreachable code rule here, or we could decide not to enable the setting. What do you prefer?

Reply:
```suggestion
             //NOTE: If the above exception is removed the break below MUST be added back.
             //break;
```
Or something like this?

Reply:
I think this is one of the very few cases where I'd be fine with a suppression and a comment explaining why.

Another possibility that would make it less scary is to remove the elses, so that they're of the form

```
if foo {
       return $x;
}
return $y;
```

and there's always a return the line above the case, so that it's obvious that all code paths don't fall through regardless of changes to the above logic.